Gopkg.toml

```diff
@@ -1,65 +1,20 @@
-[[constraint]]
-  name = "github.com/docker/go"
-  version = "1.5.1-1"
-
-[[constraint]]
-  name = "github.com/stretchr/testify"
-  version = "1.2.2"
-
-[[constraint]]
-  name = "gopkg.in/yaml.v2"
-  version = "2.2.1"
-
-[[constraint]]
-  # version imported by k8s.io/client-go @ kubernetes-1.11.2
-  name = "github.com/Azure/go-autorest"
-  revision = "1ff28809256a84bb6966640ff3d0371af82ccba4"
-
-[[constraint]]
-  name = "github.com/docker/cli"
-  revision = "f95ca8e1ba6c22c9abcdbf65e8dcc39c53958bba"
-
 [[override]]
```

Comment:
overrides without constraints are not communicated to people who depend on your library. So other people who depend upon cnab-go won't get any help from dep on these dependencies, and won't know that containerd for example should be at version 1.3.0.

Each override needs to be paired with a matching constraint for downstream consumers of the library.

Reply:
Sorry I forgot to mention the other part of the trick to let you constrain a transitive dependency with dep:

https://github.com/deislabs/porter/blob/master/pkg/dep/dep.go

It's a total hack, sorry you can blame google for killing the project before we got a chance to finish...

Anyhoo, you add a file like that in an out of the way package to get dep to consider it direct dependency. Now you can add constraints, and communicate your dependencies to your downstream consumers properly.

Reply:
I think you can get away with removing the overrides entirely with that this workaround. Give it a try. 

Then in the downstream consumer's Gokpkg.toml they can reference cnab-go without having to add a bunch of overrides and constraints because cnab-go did all the proper work up-front. That's what porter has been doing with it's mixin repos.